tests/send_specific_receipts.go

```diff
@@ -0,0 +1,99 @@
+
+	// Send two messages from Subnet A to Subnet B
+	sendCrossChainMessageInput := teleportermessenger.TeleporterMessageInput{
+		DestinationChainID: subnetBInfo.BlockchainID,
+		DestinationAddress: fundedAddress,
+		FeeInfo: teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: mockTokenAddress,
+			Amount:          big.NewInt(5),
+		},
+		RequiredGasLimit:        big.NewInt(1),
+		AllowedRelayerAddresses: []common.Address{},
+		Message:                 []byte{1, 2, 3, 4},
+	}
+
+	// Send first message from Subnet A to Subnet B with fee amount 5
+	sendCrossChainMsgReceipt, messageID1 := utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx, subnetAInfo, subnetBInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, subnetATeleporterMessenger)
+
+	// Relay message from SubnetA to SubnetB
+	network.RelayMessage(ctx, sendCrossChainMsgReceipt.BlockHash, sendCrossChainMsgReceipt.BlockNumber, subnetAInfo, subnetBInfo)
+	// Check messge delivered
+	delivered, err := subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, messageID1)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Send second message from Subnet A to Subnet B with fee amount 5
+	sendCrossChainMsgReceipt, messageID2 := utils.SendCrossChainMessageAndWaitForAcceptance(
+		ctx, subnetAInfo, subnetBInfo, sendCrossChainMessageInput, fundedAddress, fundedKey, subnetATeleporterMessenger)
+
+	// Relay message from SubnetA to SubnetB
+	network.RelayMessage(ctx, sendCrossChainMsgReceipt.BlockHash, sendCrossChainMsgReceipt.BlockNumber, subnetAInfo, subnetBInfo)
+	// Check delivered
+	delivered, err = subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, messageID2)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Relayer send specific receipts to get reward of relaying two messages
+	receipt, messageID := utils.SendSpecifiedReceiptsAndWaitForAcceptance(
+		ctx,
+		subnetAInfo.BlockchainID,
+		subnetBInfo,
+		[]*big.Int{messageID1, messageID2},
+		teleportermessenger.TeleporterFeeInfo{
+			ContractAddress: mockTokenAddress,
+			Amount:          big.NewInt(0),
+		},
+		[]common.Address{},
+		fundedAddress,
+		fundedKey,
+		subnetBTeleporterMessenger,
+	)
+
+	// Relay message from SubnetA to SubnetB
+	network.RelayMessage(ctx, receipt.BlockHash, receipt.BlockNumber, subnetBInfo, subnetAInfo)
+	// Check delivered
+	delivered, err = subnetATeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetBInfo.BlockchainID, messageID)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	// Check the reward amount. The reward amount should be 10
+	amount, err := subnetATeleporterMessenger.CheckRelayerRewardAmount(&bind.CallOpts{}, fundedAddress, mockTokenAddress)
+	Expect(err).Should(BeNil())
+	Expect(amount).Should(Equal(big.NewInt(10)))
+
```

Comment:
Can you add an additional step of sending an arbitrary message from Subnet B -> Subnet A? This message will also include the receipts that were explicitly sent in the above steps. We should check that the relayer reward amount is unchanged, even after the receipt is delivered again.

Reply:
sure. Then, will it cover `Test receiving a receipt multiple times has no effect` case?

Reply:
Yup it will! I pointed it out for completeness here, but that will also cover that test case.